src/main/java/Circle.java

```diff
@@ -0,0 +1,20 @@
+public record Circle(Point point, double radius) implements Shape {
+    public Circle {
+        if (radius < 0)
+            throw new IllegalArgumentException("Radius shouldn't be negative.");
+    }
+
+    @Override
+    public String getArea() throws IllegalStateException {
+        String resultMsg = "";
+        if (this.point == null)
+            throw new IllegalStateException("Point mustn't be null.");
+        if (this.radius == 0) {
```

Comment:
j.w.

Reply:
Tutaj moim zamiarem było "wyliczenie" getArea() dla szczególnego przypadku, gdy radius = 0 - w wymaganiach tego nie było  :D więc przyjęłam że Circle które ma radius = 0 jest nadal poprawne, a metoda jest zoptymalizowana pod tym kątem i to  wylicza :D .

Reply:
Tutaj moim zamiarem było "wyliczenie" getArea() dla szczególnego przypadku, gdy radius = 0 - w wymaganiach tego nie było  :D więc przyjęłam że Circle które ma radius = 0 jest nadal poprawne, a metoda jest zoptymalizowana pod tym kątem i to  wylicza :D .

Reply:
przedwczesna optymalizacja nie zawsze jest dobra :) poza tym porownywanie liczb typu double / float z jakas konkretna wartoscia jest zawsze obarczone ryzykiem bledu... aczkolwiek przy zerze to ma szanse zadzialac :)